tools/mo/openvino/tools/mo/main.py

```diff
@@ -308,25 +322,53 @@ def raise_ie_not_found():
         from openvino.tools.mo.front.onnx.register_custom_ops import get_front_classes
         import_extensions.load_dirs(argv.framework, extensions, get_front_classes)
 
-    return argv
 
+def check_fallback(argv : argparse.Namespace):
+    fallback_reasons = {}
 
-def prepare_ir(argv):
-    argv = arguments_post_parsing(argv)
+    if not any(deduce_framework_by_namespace(argv)): # no legacy path
+        return fallback_reasons
+
+    if argv.use_new_frontend or argv.use_legacy_frontend: # frontend chosen explicitly
+        return fallback_reasons
+
+    fallback_reasons['extensions'] = \
+        lambda argv : hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0 \
+            and argv.extensions != import_extensions.default_path() # extensions arg has default value
```

Comment:
Does it mean that I cannot activate new Extensions because it will always fallback to legacy FE?

@itikhono, please take a look. Some long time ago we discussed possibility to pass so/dll libraries with new extensions to new FE through `--extension` command line parameter. Are we going to deliver this functionality? Is it going to be a part of a separate PR?

Reply:
>> Does it mean that I cannot activate new Extensions because it will always fallback to legacy FE?
Looks like that and I think it's ok for this PR

Yes, `--extension`  with so/dll looks useful, when we want to combine extensions, e.g. ConversionExtensions, into a separate module and loading them via MO interface. 
I will create a new ticket in extension epic, but if it's already exists please let me know.

Reply:
@itikhono, agree. But we should do this soon, otherwise we won't have possibility to pass new extensions to FEs through MO.

Reply:
AR for @itikhono to continue on that.